packages/oas-resolver/index.js

```diff
@@ -80,6 +80,12 @@ function resolveAllFragment(obj, context, src, parentPath, base, options) {
                 }
                 else if (baseUrl.protocol) {
                     let newRef = url.resolve(base,obj[key]).toString();
+                    let u2 = url.parse(obj[key]);
+
+                    if(testProtocol(baseUrl.protocol, u2.protocol) === 'file:') {
+                        newRef = url.fileURLToPath('file:\\' + newRef);
```

Comment:
Just a note, fileUrlToPath was added in node 10.12.0 and node 10 went LTS at 10.13.0 so this should be ok.

Reply:
Using a backslash doesn't this mean there's only one (escaped) slash here? Should it be `\\\\` or `//` instead?

Reply:
I have changed that to use decodeURIComponent